### PR TITLE
kvstore table migration during consensus upgrade

### PR DESCRIPTION
**Pull request by ahangsu**
## Summary

The support for #3890 on `kvstore` table migration during consensus upgrade, by upgrading `accountDBVersion` from version 7 (320 rounds PR) to version 8.

### CODE REVIEW

**Review by algorandskiy — COMMENTED** [review]
I do not see where a new box table gets created

File: ledger/accountdb.go [hunk 1]
```diff
@@ -943,17 +943,25 @@ func getCatchpoint(tx *sql.Tx, round basics.Round) (fileName string, catchpoint
...
 //
 // accountsInit returns nil if either it has initialized the database
 // correctly, or if the database has already been initialized.
 func accountsInit(tx *sql.Tx, initAccounts map[basics.Address]basics.AccountData, proto config.ConsensusParams) (newDatabase bool, err error) {
-	for _, tableCreate := range accountsSchema {
-		_, err = tx.Exec(tableCreate)
-		if err != nil {
-			return
-		}
+	err = createTables(tx)
```
[the remark this answers is not in the document]
Comment (ahangsu) [reply ledger/accountdb.go:962]: ok, I will revert back the code change

File: ledger/trackerdb.go [hunk 2]
```diff
@@ -167,6 +167,13 @@ func trackerDBInitializeImpl(ctx context.Context, tx *sql.Tx, params trackerDBPa
 					tu.log.Warnf("trackerDBInitialize failed to upgrade accounts database (ledger.tracker.sqlite) from schema 5 : %v", err)
 					return
 				}
+				// NOTE wait until 320 round PR merge in master/`feature/avm-box`.
+			case 6, 7:
```
Comment (ahangsu) [inline ledger/trackerdb.go:171]: accountDBVersion 6 -> 7 is reserved for 320 rounds.

DO NOT MERGE THIS PR UNTIL master MERGED INTO feature/avm-box

File: ledger/acctupdates.go [hunk 3]
```diff
@@ -1023,6 +1023,13 @@ func (au *accountUpdates) initializeFromDisk(l ledgerForTracker, lastBalancesRou
 		return
 	}
 
+	if accountDBVersion >= int32(8) {
+		err = accountsBoxesDbQueries(au.dbs.Rdb.Handle, au.accountsq)
+		if err != nil {
+			return
+		}
+	}
```
[the remark this answers is not in the document]
Comment (ahangsu) [reply ledger/acctupdates.go:1031]: I agree that it is weird to have the condition, but the case is, if we do not apply such condition, then preparing boxes DB queries would error, for there is no kvstore in database yet.

I assume this is not the best solution in `acctupdates.go`, but for `accountdb_test.go`, where we do `accountsInitTest`, I guess we can just drop the condition and create all the tables, if there is no ledger available.

File: ledger/trackerdb.go (hunk 2 above)
Comment (ahangsu) [reply ledger/trackerdb.go]: ok, fixed after 320 in feature/avm-box and updated feature/avm-box merged in

**Review by jannotti — APPROVED** [review]
brilliant

**Pull request merged**